lib/gems/pending/util/miq-password.rb

```diff
@@ -23,21 +23,23 @@ def encrypt(str, ver = "v2", key = self.class.keys[ver])
     "#{ver}:{#{value}}"
   end
 
-  def decrypt(str, legacy = false)
-    if str.nil? || str.empty?
-      str
-    else
-      ver, enc = self.class.split(str)
-      return "" if enc.empty?
+  def decrypt(str, legacy = false, key = nil)
+    return str if str.nil? || str.empty?
+
+    ver, enc = self.class.split(str)
+    return "" if enc.empty?
+
+    ver ||= "0" # if we don't know what it is, just assume legacy
 
-      ver ||= "0" # if we don't know what it is, just assume legacy
+    key ||= begin
       key_name = (ver == "2" && legacy) ? "alt" : "v#{ver}"
```

Comment:
These few lines make me even more confused about `legacy` (especially the comment).

`legacy` seems to only be used when `ver` is 2. So why are we referring to `ver == 0` as legacy?